src/Lecture2.hs

```diff
@@ -261,7 +484,18 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval _ (Lit x) = Right x
+eval vars (Var v)
+  | isJust it = Right (fromJust it)
+  | otherwise = Left (VariableNotFound ("Error: no substution for " ++ show v))
+  where it = lookup v vars
+eval vars (Add exp1 exp2) = case (v1, v2) of
+  (Left a, _)        -> Left (VariableNotFound ("Error: Left operand isn't a Lit," ++ show a))
+  (_ , Left b)       -> Left (VariableNotFound ("Error: Right operand isn't a Lit," ++ show b))
+  (Right a, Right b) -> Right (a + b)
```

Comment:
That's a great helper function! 👏🏻 
Once you learn about Monads, you'll be able to reduce this function to a single line ✂️